src/Compilers/Core/Portable/Compilation/ConversionInfo.cs

```diff
@@ -0,0 +1,147 @@
+        /// <see cref="IConversionExpression.IsExplicit"/>.  For example, a user in C# could
+        /// write both:
+        /// 
+        /// <code>
+        ///     object o1 = ""; // and
+        ///     object o2 = (object)"";
+        /// </code>
+        /// 
+        /// These would be both be <see cref="IsWidening"/> conversions, but only the second would
+        /// have an <see cref="IConversionExpression.IsExplicit"/>.
+        /// 
+        /// <see cref="IsWidening"/> conversions can appear with other
+        /// conversion types.  For example there are <see cref="IsWidening"/> <see cref="IsReference"/>
+        /// conversions, and <see cref="IsWidening"/> <see cref="IsNumeric"/> conversions.
+        /// </summary>
+        public bool IsWidening => HasFlag(ConversionKind.IsWidening);
+
+        /// <summary>
+        /// True if and only if the conversion narrows from some broader type to some 
+        /// narrower type.  For example, a conversion from <see cref="System.Object"/> to
+        /// <see cref="System.String"/> In general, narrowing conversions appear when there is an
+        /// <see cref="IConversionExpression.IsExplicit"/> present.
+        /// 
+        /// <see cref="IsNarrowing"/> conversions can appear with other
+        /// conversion types.  For example there are <see cref="IsNarrowing"/> <see cref="IsReference"/>
+        /// conversions, and <see cref="IsNarrowing"/> <see cref="IsNumeric"/> conversions.
+        /// </summary>
+        public bool IsNarrowing => HasFlag(ConversionKind.IsNarrowing);
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit numeric conversion or explicit numeric conversion. 
+        /// </summary>
+        public bool IsNumeric => HasFlag(ConversionKind.IsNumeric);
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit nullable conversion or explicit nullable conversion.
+        /// </summary>
+        public bool IsNullable => HasFlag(ConversionKind.IsNullable);
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit reference conversion or explicit reference conversion.
+        /// </summary>
+        public bool IsReference => HasFlag(ConversionKind.IsReference);
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit default ('null' in C#, 'nothing' in VB) literal conversion.
+        /// </summary>
+        public bool IsDefaultLiteral => HasFlag(ConversionKind.IsDefaultLiteral);
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit user-defined conversion or explicit user-defined conversion.
+        /// </summary>
+        public bool IsUserDefined => HasFlag(ConversionKind.IsUserDefined);
+
+        /// <summary>
+        /// Returns the method that defines the user defined conversion, if any..
+        /// </summary>
+        public IMethodSymbol MethodSymbol { get; }
```

Comment:
>public IMethodSymbol MethodSymbol { get; } [](start = 8, length = 42)

Please put declaration of all fields and auto-properties at the top of the type, followed by constructors.